jsx/doser/edit_doser.jsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { ErrorFor, ShowError } from '../utils/validation_helper'
+import { showAlert, clearAlert } from 'utils/alert'
+import classNames from 'classnames'
+import { Field } from 'formik'
+import BooleanSelect from '../ui_components/boolean_select'
+import Cron from '../ui_components/cron'
+import Percent from '../ui_components/percent'
+
+const EditDoser = ({
+  values,
+  errors,
+  touched,
+  doser,
+  jacks,
+  submitForm,
+  isValid,
+  onBlur,
+  handleChange,
+  setFieldValue,
+  dirty,
+  readOnly
+}) => {
+  const handleSubmit = event => {
+    event.preventDefault()
+    clearAlert()
+    if (dirty === false || isValid === true) {
+      submitForm()
+    } else {
+      submitForm() // Calling submit form in order to show validation errors
+      showAlert(
```

Comment:
Use the dispatch nofication because this call is deprecated

Reply:
Do you mean addLog ? The main panel has notification component already

Reply:
The dispatch call is pretty long.  What about keeping showAlert as a convenience method instead?  This also removes the need to rewrite a bunch of the other code that is already using showAlert.

Reply:
This also places an entry in the notification log.  In this case, the alert is specifically to communicate with the user; it should not be persisted in the log.  It is not a system error, it is a validation failure.
Maybe the right answer is to refactor utils/alert to provide functionality that makes it easy to show info/warning/error and log to notification as appropriate.  I'll think on this and add comments to the issue posted earlier.

Reply:
https://github.com/reef-pi/reef-pi/blob/master/jsx/utils/ajax.js#L27 like here but there is a method to log only and no display. May i add an entry to show notification and don't log it?

Reply:
See Issue #590 for new comments and additional conversation.